### ISSUE TITLE
NOCHECKIN does not work as intended.

### ISSUE THREAD

**Issue by hgreving2304** [DynamoRIO/dynamorio#3588]
https://github.com/DynamoRIO/dynamorio/wiki/Code-Style suggests to mark temporary code unsuited for committing with NOCHECKIN. However, if you do that, the builds fail, making it unsuitable for testing e.g. on Travis.

If NOCHECKIN would prevent the code being able to merge (pre-commit hook on Travis?), it would make sense. Otherwise I'm not sure how the NOCHECKIN hook could be used for as long as we're using Travis.

**Comment by derekbruening** [comment]
Most testing should be done *before* the Travis CI tests: Travis should not be the primary testing step in the workflow but a final sanity check.

**Comment by hgreving2304** [comment]
Of course, but in certain cases things are not reproducible locally (e.g. in my case it was osx). Why would I use NOCHECKIN locally where nobody can see it?

**Comment by derekbruening** [comment]
> Why would I use NOCHECKIN locally where nobody can see it?

So you don't push code upstream that was for local debugging!

**Comment by hgreving2304** [comment]
Ok. I guess in pull/3585 it wasn't clear to you that I made the temporary changes *for* running on Travis, because I couldn't reproduce locally (osx).

Maybe another way for NOCHECKIN would be to add a test that checks for NOCHECKIN in the code and fails.

With respect to this issue, if this *is* the intention of the NOCHECKIN hook, close it.

**Comment by derekbruening** [comment]
Agreed that it does not work directly when putting in temporary code to diagnose CI issues.

> Maybe another way for NOCHECKIN would be to add a test (not the build) that checks for NOCHECKIN in the code and fails.

This seems to be the key: the problem is that it blocks everything up front.  A test-level failure would still keep it red while allowing diagnosing other build/test issues.
